FirebaseCombineSwift/Tests/Unit/Auth/UserTests.swift

```diff
@@ -0,0 +1,440 @@
+
+    override func verifyPassword(_ request: FIRVerifyPasswordRequest,
+                                 callback: @escaping FIRVerifyPasswordResponseCallback) {
+      let response = MockVerifyPasswordResponse()
+      callback(response, nil)
+    }
+
+    override func getAccountInfo(_ request: FIRGetAccountInfoRequest,
+                                 callback: @escaping FIRGetAccountInfoResponseCallback) {
+      XCTAssertEqual(request.apiKey, Credentials.apiKey)
+      XCTAssertEqual(request.accessToken, accessToken)
+
+      let response = MockGetAccountInfoResponse()
+      response.providerCredentials = providerCredentials
+      callback(response, nil)
+    }
+  }
+
+  func testlinkAndRetrieveDataSuccess() {
+    let facebookCredentials = ProviderCredentials(
+      providerID: FacebookAuthProviderID,
+      federatedID: "FACEBOOK_ID",
+      displayName: "Facebook Doe",
+      providerIDToken: nil,
+      providerAccessToken: "FACEBOOK_ACCESS_TOKEN",
+      email: UserTests.email
+    )
+
+    let authBackend = MockAuthBackend()
+    authBackend.providerCredentials = facebookCredentials
+    FIRAuthBackend.setBackendImplementation(authBackend)
+
+    var cancellables = Set<AnyCancellable>()
+    let userSignInExpectation = expectation(description: "User associated from a third-party")
+
+    let facebookCredential = FacebookAuthProvider
+      .credential(withAccessToken: facebookCredentials.providerAccessToken)
+
+    // when
+    Auth.auth()
+      .signIn(with: facebookCredential)
+      .flatMap { [weak authBackend] (authResult) -> Future<AuthDataResult, Error> in
+        XCTAssertEqual(authResult.additionalUserInfo?.profile,
+                       UserTests.googleProfile as [String: NSString])
+        XCTAssertEqual(authResult.additionalUserInfo?.username,
+                       UserTests.userName)
+        XCTAssertEqual(authResult.additionalUserInfo?.providerID,
+                       FacebookAuthProviderID)
+
+        let googleCredentials = ProviderCredentials(
+          providerID: GoogleAuthProviderID,
+          federatedID: "GOOGLE_ID",
+          displayName: "Google Doe",
+          providerIDToken: "GOOGLE_ID_TOKEN",
+          providerAccessToken: "GOOGLE_ACCESS_TOKEN",
+          email: UserTests.googleEmail
+        )
+
+        authBackend?.providerCredentials = googleCredentials
+
+        let linkGoogleCredential = GoogleAuthProvider.credential(
+          withIDToken: googleCredentials.providerIDToken!,
+          accessToken: googleCredentials.providerAccessToken
+        )
+        return authResult.user
+          .link(with: linkGoogleCredential)
+      }
+      .sink { completion in
+        switch completion {
+        case .finished:
+          print("Finished")
+        case let .failure(error):
+          XCTFail("💥 Something went wrong: \(error)")
+        }
+      } receiveValue: { linkAuthResult in
+        let user = linkAuthResult.user
+        XCTAssertEqual(Auth.auth().currentUser, user)
```

Comment:
The original test has a few comments that explain what is being verified (see https://cs.opensource.google/firebase-sdk/firebase-ios-sdk/+/master:FirebaseAuth/Tests/Unit/FIRUserTests.m;l=1910). Let's add these here as well.